.github/workflows/browser-tests.yml

```diff
@@ -0,0 +1,154 @@
+
+            - name: Set up whole project using the tested dependency
+              run: ./vendor/bin/prepare_project_edition.sh ${{ inputs.project-edition }} ${{ inputs.project-version }} ${{ inputs.setup }} ${{ inputs.php-image }}
+
+            - if: inputs.multirepository
+              name: Set up multirepository build
+              run: $HOME/build/project/vendor/ezsystems/behatbundle/bin/.travis/prepare_multirepository_setup.sh
+
+            - if: inputs.test-setup-phase-1 != ''
+              name: Run first phase of tests setup
+              run: |
+                  cd ${HOME}/build/project
+                  docker-compose --env-file=.env exec -T --user www-data app sh -c "vendor/bin/ezbehat ${{ inputs.test-setup-phase-1 }}"
+
+            - if: inputs.test-setup-phase-2 != ''
+              name: Run second phase of tests setup
+              run: |
+                  cd ${HOME}/build/project
+                  docker-compose --env-file=.env exec -T --user www-data app sh -c "composer run post-install-cmd"
+                  docker-compose --env-file=.env exec -T --user www-data app sh -c "vendor/bin/ezbehat ${{ inputs.test-setup-phase-2 }}"
+
+            - name: Run tests
+              run: |
+                  cd ${HOME}/build/project
+                  docker-compose --env-file=.env exec -T --user www-data app sh -c "vendor/bin/ezbehat ${{ inputs.test-suite }}"
+
+            - if: always()
+              name: Upload tests report
+              run: |
+                  cd ${HOME}/build/project
+                  vendor/bin/ezreport
+
+    report-results:
+        runs-on: ubuntu-latest
+        timeout-minutes: 3
+        if: always()
+        needs: browser-tests
+        steps:
+            - name: Create Slack failure message
+              run: |
+                  echo "SLACK_PAYLOAD={\"blocks\": [{\"type\": \"section\",\"text\": {\"type\": \"mrkdwn\",\"text\": \":x: Browser tests for $GITHUB_REPOSITORY repository have failed.\nCommiter: $GITHUB_ACTOR \nBranch: $GITHUB_REF_NAME \nDetails: $GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID \"}}]}" >> $GITHUB_ENV
+            - if: ${{ needs.browser-tests.result == 'success' }}
+              name: Create Slack success message
+              run: |
+                  echo "SLACK_PAYLOAD={\"blocks\": [{\"type\": \"section\",\"text\": {\"type\": \"mrkdwn\",\"text\": \":white_check_mark: Browser tests for $GITHUB_REPOSITORY repository have passed.\nCommiter: $GITHUB_ACTOR \nBranch: $GITHUB_REF_NAME \nDetails: $GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID \"}}]}" >> $GITHUB_ENV
+            - if: github.event_name != 'pull_request'
+              name: Send notification about workflow result
+              uses: slackapi/slack-github-action@v1.16.0
+              with:
+                  payload: ${{ env.SLACK_PAYLOAD }}
+              env:
+                  SLACK_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
```

Comment:
We need to create this secret.

Can we be sure it will not be somehow leaked through the action?

Reply:
This is an official Slack action: https://github.com/slackapi/slack-github-action , so we shouldn't worry that the action starts doing something malicious (well, it can, but it's less likely than for an unofficial action).

The only other danger I see is that it posts the webhook url somewhere (for example during error handling in the build log, but then it should be masked automatically because it's a secret:
```
Warning: GitHub automatically redacts secrets printed to the log, but you should avoid printing secrets to the log intentionally.
```
(from: https://docs.github.com/en/actions/security-guides/encrypted-secrets)

Creating the webhook using an app requires to choose the channel it can post messages to, so even if this secret is leaked someone would only be able to post messages to that one specific channel.

Is there anything else we should think about?